src/sysext/sysext.c

```diff
@@ -2208,6 +2209,46 @@ static int vl_method_refresh(sd_varlink *link, sd_json_variant *parameters, sd_v
         return sd_varlink_reply(link, NULL);
 }
 
+static int parse_version(const char *filename, char **version) {
```

Comment:
Missing assertion for `filename`,
Please rename `version` -> `ret`, and also add assertion for it.